pkgs/tools/filesystems/glusterfs/default.nix

```diff
@@ -36,18 +54,68 @@ rec {
 
   makeFlags = "DESTDIR=$(out)";
 
-  preInstall = ''
-    substituteInPlace api/examples/Makefile --replace '$(DESTDIR)' $out
-    substituteInPlace geo-replication/syncdaemon/Makefile --replace '$(DESTDIR)' $out
-    substituteInPlace geo-replication/syncdaemon/Makefile --replace '$(DESTDIR)' $out
-    substituteInPlace xlators/features/glupy/examples/Makefile --replace '$(DESTDIR)' $out
-    substituteInPlace xlators/features/glupy/src/Makefile --replace '$(DESTDIR)' $out
-    '';
+  enableParallelBuilding = true;
 
   postInstall = ''
     cp -r $out/$out/* $out
     rm -r $out/nix
+    '';
+
+  # Below we run Python programs. That generates .pyc/.pyo files.
+  # By default they are indeterministic because such files contain time stamps
+  # (see https://nedbatchelder.com/blog/200804/the_structure_of_pyc_files.html).
+  # So we use the same environment variables as in
+  #   https://github.com/NixOS/nixpkgs/blob/249b34aadca7038207492f29142a3456d0cecec3/pkgs/development/interpreters/python/mk-python-derivation.nix#L61
+  # to make these files deterministic.
+  DETERMINISTIC_BUILD = 1;
+  PYTHONHASHSEED = 0;
+
+  postFixup = ''
     wrapProgram $out/sbin/mount.glusterfs --set PATH "${stdenv.lib.makeBinPath [ coreutils gnused attr gnugrep which]}"
+
+    # glusterd invokes `gluster` and other utilities when telling other glusterd nodes to run commands,
+    # for example for `peer_georep-sshkey` key generation, so `$out/bin` is needed in the PATH.
+    # It also invokes executable Python scripts in `$out/libexec/glusterfs`, which is why we set up PYTHONPATH accordingly.
+    # Because the user may invoke `gluster` directly, we also set up the paths for that executable.
+    wrapProgram $out/bin/glusterd --set PATH "${stdenv.lib.makeBinPath [ openssh ]}:$out/bin" --set PYTHONPATH "$(toPythonPath $out)"
```

Comment:
instead of everywhere putting `$(toPythonPath $out):$out/libexec/glusterfs` you may instead do `PYTHONPATH=$(toPythonPath $out):$out/libexec/glusterfs` and reuse that. I think it becomes more readable that way.